sound/soc/sof/sof-acpi-dev.c

```diff
@@ -149,13 +149,8 @@ static void sof_acpi_fw_cb(const struct firmware *fw, void *context)
 	}
 
 	/* register PCM and DAI driver */
-	priv->pdev_pcm =
-		platform_device_register_data(dev, "sof-audio", -1,
-					      sof_pdata, sizeof(*sof_pdata));
-	if (IS_ERR(priv->pdev_pcm)) {
-		dev_err(dev, "Cannot register device sof-audio. Error %d\n",
-			(int)PTR_ERR(priv->pdev_pcm));
-	}
+	sof_create_platform_device(priv);
```

Comment:
Need to check return value.

Reply:
this callback function actually will be removed in the next commit.

Reply:
but let me remove this change here in the update versions.

Reply:
@keyonjie feel free to squash these patches per file if it makes things easier. They will be upstreamed as a single patch per file.